Forgot Password

JimtheBurger opened the pull request:
# Proposed Changes

* Has a Forgot Password flow
* Has relevant APIs that do not touch the database.

Fixes #114  .

**IMPORTANT:** You must add these to your `.env` variables to have the email check functionality work as:
```
SMTP_HOST=smtp.gmail.com
SMTP_PORT=587
SMTP_USER=adversarialapps@gmail.com
SMTP_PASSWORD=zqja etil gwme utmd
SMTP_FROM_EMAIL=adversarialapps@gmail.com
```
(These have already been added to Heroku)

JimtheBurger committed:
Revert "Reset Token is sending!"

This reverts commit 1c263f4e41dcde56eaae9e743cb7c1e11026d199.

JimtheBurger committed:
Revert "Revert "Reset Token is sending!""

This reverts commit 5992804f17c4ab221e340a12ada597c4099b1c1f.

JimtheBurger committed:
Revert "Revert "first successful forgot password flow!""

This reverts commit 2cd197905feed9eda7bb36bd869d3929aa9be2ce.

JimtheBurger committed:
pretty clean work flow

JimtheBurger committed:
fixes lint

JimtheBurger committed:
Merge remote-tracking branch 'origin/development' into 114-forgot-password

JimtheBurger committed:
re-added route.ts

JimtheBurger committed:
comment fixes

JimtheBurger committed:
fixed black

JimtheBurger committed:
fixes isort

JimtheBurger committed:
fixes isort, attempt 2

JimtheBurger committed:
ran isort

JimtheBurger committed:
added nodemailer to package.json

JimtheBurger committed:
better requirements.txt

JimtheBurger committed:
Merge remote-tracking branch 'origin/development' into 114-forgot-password

JimtheBurger committed:
fixed black

chrisrod1307 reviewed (changes requested):
Nice work! I recieved the forgot password token on my email, used it to create a new password and was able to sign in with my new password. Old password is invalid now and replaced with new password. Just two small comments:
* Is it possible to make the token shorter? If not, its perfectly fine but I think if we make it 8-10 random characters it should be enough.
* There is no password validation when entering the new password so I could potentially enter a super unsafe password (ex. 1234). We already have password validation logic in `signup/page.tsx` so you can just copy that over. 

Good to merge in after those 2 changes!

JimtheBurger committed:
added forgot password validation

JimtheBurger commented:
> Nice work! I recieved the forgot password token on my email, used it to create a new password and was able to sign in with my new password. Old password is invalid now and replaced with new password. Just two small comments:
> 
> * Is it possible to make the token shorter? If not, its perfectly fine but I think if we make it 8-10 random characters it should be enough.
> * There is no password validation when entering the new password so I could potentially enter a super unsafe password (ex. 1234). We already have password validation logic in `signup/page.tsx` so you can just copy that over.
> 
> Good to merge in after those 2 changes!

**Is it possible to make the token shorter? If not, its perfectly fine but I think if we make it 8-10 random characters it should be enough.**
I just fixed this, it directly copies the password validation from creating a password in sign-up

**There is no password validation when entering the new password so I could potentially enter a super unsafe password (ex. 1234). We already have password validation logic in `signup/page.tsx` so you can just copy that over.**
I would need to change the token generation if this is the case, it appears I would need to change libraries to do this. I think it's worth putting in a ticket for it for a future change, but don't think we need to extend this tickets lifespan much farther. Willing to be wrong on this though.

JimtheBurger committed:
Merge remote-tracking branch 'origin/development' into 114-forgot-password

JimtheBurger committed:
fixed requirements
